src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptionsSetup.cs

```diff
@@ -52,8 +57,47 @@ public void Configure(LanguageWorkerOptions options)
                 return;
             }
 
-            var configFactory = new RpcWorkerConfigFactory(_configuration, _logger, SystemRuntimeInformation.Instance, _environment, _metricsLogger, _workerProfileManager);
+            // Use the latest configuration from the ScriptHostManager if available.
+            // After specialization, the ScriptHostManager will have the latest IConfiguration reflecting additional configuration entries added during specialization.
+            var configuration = _configuration;
+            if (_scriptHostManager is IServiceProvider scriptHostManagerServiceProvider)
+            {
+                var latestConfiguration = scriptHostManagerServiceProvider.GetService<IConfiguration>();
+                if (latestConfiguration is not null)
+                {
+                    configuration = new ConfigurationBuilder()
+                        .AddConfiguration(_configuration)
+                        .AddConfiguration(latestConfiguration)
+                        .Build();
+                }
+            }
+
+            var configFactory = new RpcWorkerConfigFactory(configuration, _logger, SystemRuntimeInformation.Instance, _environment, _metricsLogger, _workerProfileManager);
             options.WorkerConfigs = configFactory.GetConfigs();
         }
     }
+
+    /// <summary>
+    /// This implementation of IPostConfigureOptions validates that LanguageWorkerOptions are not configured within the JobHost scope.
+    /// LanguageWorkerOptions should be forwarded from the parent scope.
+    /// Triggers a debug failure and logs a message if unexpected configuration is detected.
+    /// </summary>
+    internal class JobHostLanguageWorkerOptionsSetup : IPostConfigureOptions<LanguageWorkerOptions>
+    {
+        private readonly ILoggerFactory _loggerFactory;
+
+        public JobHostLanguageWorkerOptionsSetup(ILoggerFactory loggerFactory)
+        {
+            _loggerFactory = loggerFactory;
+        }
+
+        public void PostConfigure(string name, LanguageWorkerOptions options)
+        {
+            var message = "Unexpected configuration of LanguageWorkerOptions from the JobHost scope. LanguageWorkerOptions should be forwarded from the parent scope with no additional configuration.";
+            Debug.Fail(message);
+
+            var logger = _loggerFactory.CreateLogger("Host.LanguageWorkerConfig");
```

Comment:
Can we create a type-based logger here (`CreateLogger<JobHostLanguageWorkerOptionsSetup>()`)? We're trying to get away from these "Host..." ones so let's not try to add more.

Reply:
ping on this one...

Reply:
nevermind, i'm dumb. wrong iteration :)